includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -755,50 +767,79 @@ public function display_payment_request_button_separator_html() {
 	}
 
 	/**
-	 * Whether payment button html should be rendered on the Cart
+	 * Returns true if Payment Request Buttons are supported on the current page, false
+	 * otherwise.
 	 *
-	 * @since  4.4.1
-	 * @return boolean
+	 * @since   5.3.0
+	 * @version 5.3.0
+	 * @return  boolean  True if PRBs are supported on current page, false otherwise
 	 */
-	private function should_show_payment_button_on_cart() {
+	public function should_show_payment_request_button() {
+		global $post;
+
+		// If keys are not set bail.
+		if ( ! $this->are_keys_set() ) {
+			WC_Stripe_Logger::log( 'Keys are not set correctly.' );
+			return false;
+		}
+
+		// If no SSL bail.
+		if ( ! $this->testmode && ! is_ssl() ) {
+			WC_Stripe_Logger::log( 'Stripe Payment Request live mode requires SSL.' );
+			return false;
+		}
+
 		// Not supported when user isn't authenticated and authentication is required.
 		if ( ! is_user_logged_in() && $this->is_authentication_required() ) {
 			return false;
 		}
 
-		if ( ! apply_filters( 'wc_stripe_show_payment_request_on_cart', true ) ) {
+		// Don't show if on the cart or checkout page, or if page contains the cart or checkout
+		// shortcodes, with items in the cart that aren't supported.
+		if (
+			( is_cart() || is_checkout() || WC_Stripe_Helper::has_cart_or_checkout_shortcode_on_current_page() )
```

Comment:
Note: This will be redundant too after rebase. We can either remove `has_cart_or_checkout_on_current_page` from trunk and always use `is_cart` & `is_checkout` explicitly or the other way around.

Reply:
Fixed in 1f0eb7a7f342658d1efeb44a67fd13bd4f2a8d32